metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,660 @@
+    }
+
+    def create_scan_job(self, workspace_id: str) -> str:
+        """
+        Create scan job on PowerBI for the workspace
+        """
+        request_body = {"workspaces": [workspace_id]}
+
+        scan_create_endpoint = AdminAPIResolver.API_ENDPOINTS[Constant.SCAN_CREATE]
+        scan_create_endpoint = scan_create_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL
+        )
+
+        logger.debug(
+            f"Creating metadata scan job, request body {request_body}",
+        )
+
+        res = self._request_session.post(
+            scan_create_endpoint,
+            data=request_body,
+            params={
+                "datasetExpressions": True,
+                "datasetSchema": True,
+                "datasourceDetails": True,
+                "getArtifactUsers": True,
+                "lineage": True,
+            },
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        res.raise_for_status()
+        # Return scan_id of Scan created for the given workspace
+        scan_id = res.json()["id"]
+
+        logger.info(f"Scan id({scan_id})")
+
+        return scan_id
+
+    @staticmethod
+    def _calculate_max_retry(minimum_sleep: int, timeout: int) -> int:
+        if timeout < minimum_sleep:
+            logger.info(
+                f"Setting timeout to minimum_sleep time {minimum_sleep} seconds"
+            )
+            timeout = minimum_sleep
+
+        return timeout // minimum_sleep
+
+    def wait_for_scan_to_complete(self, scan_id: str, timeout: int) -> Any:
+        """
+        Poll the PowerBi service for workspace scan to complete
+        """
+        minimum_sleep = 3
+        max_retry: int = AdminAPIResolver._calculate_max_retry(minimum_sleep, timeout)
+        logger.info(f"Max trial {max_retry}")
+
+        scan_get_endpoint = AdminAPIResolver.API_ENDPOINTS[Constant.SCAN_GET]
+        scan_get_endpoint = scan_get_endpoint.format(
+            POWERBI_ADMIN_BASE_URL=DataResolverBase.ADMIN_BASE_URL, SCAN_ID=scan_id
+        )
+        logger.debug(f"Hitting URL={scan_get_endpoint}")
+        retry = 1
+        while True:
+            logger.info(f"retry = {retry}")
+            res = self._request_session.get(
+                scan_get_endpoint,
+                headers={Constant.Authorization: self.get_access_token()},
+            )
+            if res.status_code != 200:
+                message = f"API({scan_get_endpoint}) return error code {res.status_code} for scan id({scan_id})"
+                logger.warning(message)
+                raise ConnectionError(message)
+
+            if res.json()["status"].upper() == "SUCCEEDED":
+                logger.info(f"Scan result is available for scan id({scan_id})")
+                return True
+
+            if retry == max_retry:
+                logger.warning("Max retry reached. Scan job result is not available")
```

Comment:
Max retry reached when polling for scan job (lineage) result. Scan job is not available! Try increasing your max retry using config option <param name here>

Reply:
done